server/controllers/refer/referralsController.ts

```diff
@@ -0,0 +1,41 @@
+import type { Request, Response, TypedRequestHandler } from 'express'
+import createError from 'http-errors'
+
+import type { CourseService, OrganisationService } from '../../services'
+import { courseUtils, typeUtils } from '../../utils'
+
+export default class ReferralsController {
+  constructor(
+    private readonly courseService: CourseService,
+    private readonly organisationService: OrganisationService,
+  ) {}
+
+  start(): TypedRequestHandler<Request, Response> {
+    return async (req: Request, res: Response) => {
+      typeUtils.assertHasUser(req)
+
+      const course = await this.courseService.getCourse(req.user.token, req.params.courseId)
+      const courseOffering = await this.courseService.getOffering(
+        req.user.token,
+        req.params.courseId,
+        req.params.courseOfferingId,
+      )
+      const organisation = await this.organisationService.getOrganisation(req.user.token, courseOffering.organisationId)
+
+      if (!organisation) {
```

Comment:
I don't think we've got these yet, but it'd be good to add tests for 404ing in controllers.

Reply:
Can't remember if I had trouble with this when trying the flag stuff but I'll give this a go tomorrow

Reply:
I've added one [here](https://github.com/ministryofjustice/hmpps-accredited-programmes-ui/pull/143/commits/9d525a6feaedd8dc70984b0608bc3481f8e28f78) if it's useful

Reply:
I've added a commit for this: https://github.com/ministryofjustice/hmpps-accredited-programmes-ui/pull/142/commits/7bfb50a6e9fa0582da2fa9bb88999730e3b07dfa

I found that adding the object after `404` in `createError` made no difference to whether the test passed - you could put any object in there and it would be happy - so I simplified it a bit from your example

Reply:
Ah interesting, good to know - thanks for updating this.